### ISSUE TITLE
[Doc] Add implementation status / matrix

### ISSUE THREAD

**Issue by rok** [rok/test-parquet-format#157]
In Apache Arrow we have a documentation page listed the feature status for various implementations of Arrow: https://arrow.apache.org/docs/status.html

It could be nice to have a similar page for the main Parquet implementations (at least Java, C++, Rust).

The main downside is that it needs to be kept up to date.

**Reporter**: [Antoine Pitrou](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=apitrou) / @pitrou
#### PRs and other links:
- [GitHub Pull Request #34](https://github.com/apache/parquet-site/pull/34)

<sub>**Note**: *This issue was originally created as [PARQUET-2310](https://issues.apache.org/jira/browse/PARQUET-2310). Please see the [migration documentation](https://github.com/apache/arrow/issues/14542) for further details.*</sub>

**Comment by rok** [comment]
[Antoine Pitrou](https://issues.apache.org/jira/browse/PARQUET-2310?#comment-17733099) / @pitrou:
cc @wgtmac @gszadovszky @alippai

**Comment by rok** [comment]
[Antoine Pitrou](https://issues.apache.org/jira/browse/PARQUET-2310?#comment-17733100) / @pitrou:
This was originally proposed in https://github.com/apache/arrow/pull/36027

**Comment by rok** [comment]
[Andrew Lamb](https://issues.apache.org/jira/browse/PARQUET-2310?#comment-17846304) / @alamb:
Examples from `[~crepererum]` on <https://github.com/apache/parquet-site/pull/59#discussion_r1599844055>
- <https://jpeg.org/jpegxl/software.html> (the list format is good, the fact that there's only a single implementation is not)
- <https://paseto.io/>
- <https://autocrypt.org/dev-status.html>

**Comment by rok** [comment]
[Andrew Lamb](https://issues.apache.org/jira/browse/PARQUET-2310?#comment-17846306) / @alamb:
I am interested in helping this issue. Once <https://github.com/apache/parquet-site/pull/53> is merged I think we can begin work. Per <https://github.com/apache/parquet-site/pull/34#issuecomment-2098269154> I think @alippai  plans to draft it
